core/store/src/db/splitdb.rs

```diff
@@ -123,13 +143,13 @@ impl Database for SplitDB {
     /// The returned iterator will iterate through items in both the cold store
     /// and the hot store. The items will be unique and sorted.
     fn iter_prefix<'a>(&'a self, col: DBCol, key_prefix: &'a [u8]) -> DBIterator<'a> {
-        if !col.is_cold() {
+        if !col.is_cold() || self.cold.is_none() {
```

Comment:
Is this a breaking change? I'm guessing cloud does not support iteration. How can we make sure that this doesn't break any use cases?

Reply:
looking at the codebase `get_outcomes_by_id` is interesting - it might be used by tx_status and it relies on the iter_prefix. We may need to keep any column that uses iter* in cold rather than cloud. There are exceptions like garbage collection that only needs hot or state sync that only needs snapshot (of the hot). I'm afraid it'll need to be a one by one investigation.

Reply:
I did a one by one investigation in the past, searching for references to iter methods.
Now I did it by searching for references to cold columns. Still looks fine - [here](https://docs.google.com/document/d/1_BsVHATijonzrtRKgBwvawkz87j902dpxCCQdsSdFeo/edit?usp=sharing) is a summary.
So there are a few cold columns that might get called with iter but we can deal with them. In this PR, I only handle `DbCol::Block`, so ignored them for now.